brave-core/src/main/java/com/github/kristofa/brave/HttpSpanCollector.java

```diff
@@ -0,0 +1,198 @@
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ScheduledExecutorService;
+import okio.Buffer;
+import okio.GzipSink;
+import org.apache.thrift.TException;
+import org.apache.thrift.protocol.TBinaryProtocol;
+import org.apache.thrift.protocol.TList;
+import org.apache.thrift.protocol.TType;
+import org.apache.thrift.transport.TTransport;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+/**
+ * SpanCollector which submits spans to Zipkin, using its {@code POST /spans} endpoint.
+ */
+public final class HttpSpanCollector implements SpanCollector, Flushable, Closeable {
+
+  private final String url;
+  private final SpanCollectorMetricsHandler metrics;
+  private final BlockingQueue<Span> pending = new LinkedBlockingQueue<>(1000);
+  @Nullable // for testing
+  private final Flusher flusher;
+
+  /**
+   * Create a new instance with default configuration.
+   *
+   * @param baseUrl URL of the zipkin query server instance. Like: http://localhost:9411/
+   * @param metrics Gets notified when spans are accepted or dropped. If you are not interested in
+   * these events you can use {@linkplain EmptySpanCollectorMetricsHandler}
+   */
+  public static HttpSpanCollector create(String baseUrl, SpanCollectorMetricsHandler metrics) {
+    return new HttpSpanCollector(baseUrl, metrics, 1);
+  }
+
+  // Visible for testing. Ex when tests need to explicitly control flushing, set interval to 0.
+  HttpSpanCollector(String baseUrl, SpanCollectorMetricsHandler metrics, int flushInterval) {
+    this.url = baseUrl + (baseUrl.endsWith("/") ? "" : "/") + "api/v1/spans";
+    this.metrics = metrics;
+    this.flusher = flushInterval > 0 ? new Flusher(this, flushInterval) : null;
+  }
+
+  /**
+   * Queues the span for collection, or drops it if the queue is full.
+   *
+   * @param span Span, should not be <code>null</code>.
+   */
+  @Override
+  public void collect(Span span) {
+    metrics.incrementAcceptedSpans(1);
+    if (!pending.offer(span)) {
+      metrics.incrementDroppedSpans(1);
+    }
+  }
+
+  /**
+   * Calling this will flush any pending spans to the http transport on the current thread.
+   */
+  @Override
+  public void flush() {
+    if (pending.isEmpty()) return;
+    Queue<Span> drained = new ArrayDeque<>(pending.size());
+    pending.drainTo(drained);
+    if (drained.isEmpty()) return;
+
+    // Thrift-encode the spans for transport
+    int spanCount = drained.size();
+    BufferTransport buffer = new BufferTransport();
+    try {
+      TBinaryProtocol oprot = new TBinaryProtocol(buffer);
+      oprot.writeListBegin(new TList(TType.STRUCT, drained.size()));
+      while (!drained.isEmpty()) {
+        drained.remove().write(oprot);
+      }
+      oprot.writeListEnd();
+    } catch (TException e) {
+      metrics.incrementDroppedSpans(spanCount);
+      return;
+    }
+
+    // Compress the thrifts, as they are usually full of strings
+    Buffer gzippedThrifts = new Buffer();
```

Comment:
ps in the future, I'd likely rework this class to use a soft size boundary. Ex. add spans until we hit a certain payload length. I've other things that are on the mind right now, so punting the idea as I'm sure it would take some number of hours above 2 to get it right.